Introducing structured logging with Serilog

NikolaVetnic opened the pull request:
Completing the [DEV-4775 Structured Logging](https://bildungsraum.vdivde-it.de/jira/browse/DEV-4775) task:
1. Added the `Serilog.Formatting.Compact` dependency required to output JSON objects in Console,
2. Removed the `outputTemplate` entry from the Console sink settings in `ConsumerApi\appsettings.override.json` and added the `formatter` setting,
3. All logger entries of format `_logger.LogInformation($"Text {value}");` were changed to `_logger.LogInformation("Text {value}", value);`.

No tests were written or modified to accompany this task.

tnotheis reviewed (changes requested):
Can you paste some sample output in a comment please?

And finally, I have 2 major requests:
* Could you as well update all the other log statements (e.g. in the SanityCheck.* projects), so that all log statements are correct? You can use the following regex to find all of them (but not the ones you already fixed): 
```regex
logger.Log[^(]+\(\s*\$
```
* Could you also change all log statements so that the placeholders are wrapped in single quotes? So instead of 
```c#
logger.Log("Text {value}", value);
```
it should be
```c#
logger.Log("Text '{value}'", value);
```

NikolaVetnic commented:
@tnotheis of course, I will make the requested changes immediately.

NikolaVetnic commented:
Sample output:
```JSON
{
   "@t":"2023-09-25T11:21:53.3635310Z",
   "@mt":"Loaded hosting startup assembly {assemblyName}",
   "@l":"Debug",
   "assemblyName":"ConsumerApi",
   "EventId":{
      "Id":13,
      "Name":"HostingStartupAssemblyLoaded"
   },
   "SourceContext":"Microsoft.AspNetCore.Hosting.Diagnostics"
}

{
   "@t":"2023-09-25T11:21:53.3640676Z",
   "@mt":"Hosting started",
   "@l":"Debug",
   "EventId":{
      "Id":2,
      "Name":"Started"
   },
   "SourceContext":"Microsoft.Extensions.Hosting.Internal.Host"
}
```

Multi-line log statements that were missed in the first pass are now updated as well.

Modified all the log statements so that the placeholders are wrapped in single quotes (e.g. `logger.Log("Text '{value}'", value);`) using the following regex as a search term `(logger\.Log\w*\("[^{']*?)(?<!')\{([^}]*?)\}(?!')(.*?",)` and this one as the replacements value `$1'{$2}'$3`.

Removed the `ApplicationInsights` sink.

NikolaVetnic commented on ConsumerApi/appsettings.override.json:
```diff
@@ -136,15 +136,15 @@
+                  "outputTemplate": "[{Component}|{MachineName}|{ThreadId}] {Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level:u3}] <{SourceContext}> {Message:lj}{NewLine}{Exception}",
+                  "formatter": "Serilog.Formatting.Compact.CompactJsonFormatter, Serilog.Formatting.Compact"
                 }
             }
```
Done, check the latest commits.

NikolaVetnic commented on Modules/Quotas/src/Quotas.Application/IntegrationEvents/Incoming/TierQuotaDefinitionDeleted/TierQuotaDefinitionDeletedIntegrationEventHandler.cs:
```diff
@@ -21,13 +21,13 @@ public class TierQuotaDefinitionDeletedIntegrationEventHandler : IIntegrationEve
     public async Task Handle(TierQuotaDefinitionDeletedIntegrationEvent @event)
     {
-        _logger.LogTrace($"Handling '{nameof(TierQuotaDefinitionDeletedIntegrationEvent)}' ... ");
+        _logger.LogTrace("Handling '{tierQuotaDefinitionDeletedIntegrationEvent}' ... ", nameof(TierQuotaDefinitionDeletedIntegrationEvent));
```
Renamed the placeholder, check the latest commit.

tnotheis commented:
> Sample output:
> 
> ```json
> {
>    "@t":"2023-09-25T11:21:53.3635310Z",
>    "@mt":"Loaded hosting startup assembly {assemblyName}",
>    "@l":"Debug",
>    "assemblyName":"ConsumerApi",
>    "EventId":{
>       "Id":13,
>       "Name":"HostingStartupAssemblyLoaded"
>    },
>    "SourceContext":"Microsoft.AspNetCore.Hosting.Diagnostics"
> }
> 
> {
>    "@t":"2023-09-25T11:21:53.3640676Z",
>    "@mt":"Hosting started",
>    "@l":"Debug",
>    "EventId":{
>       "Id":2,
>       "Name":"Started"
>    },
>    "SourceContext":"Microsoft.Extensions.Hosting.Internal.Host"
> }
> ```

Sorry, I did not express myself correctly. I wanted to see some custom logs. The ones you sent me are produced by the framework. Clear your output window after the application has started, then send a request to a random endpoint and afterwards copy the content of the output window. Maybe you should create a file with the content because it will probably be very long. ;)

NikolaVetnic commented:
Here is a console output sample: https://jssoftgmbh-my.sharepoint.com/:t:/g/personal/nikola_vojislav-vetnic_js-soft_rs/Ef5FmwxAuf5Im304rDKeoUYBaawzFooUnPJovBQQuWt0Xg?email=Timo.Notheisen%40js-soft.com&e=FLxh5D
